synapse/storage/databases/main/deviceinbox.py

```diff
@@ -664,21 +664,28 @@ def _remove_deleted_devices_from_device_inbox_txn(
             due to a single device having lots of device messages.
             """
 
-            last_stream_id = progress.get("stream_id", 0)
+            txn.execute("SELECT max(stream_id) FROM device_inbox")
+            row = txn.fetchone()
+            if row is None or row[0] is None:
```

Comment:
any idea what could cause `row[0]` to be `None`?

Reply:
oh wait, `row[0] is None` if the table is empty. In which case, my question becomes: any idea what could cause `row` to be `None`?

Reply:
I did it to stop mypy from complaining, but I'm just going to replace it with an ignore and a comment explaining the ignore.